community/blog/2022-09 New types of request tracking in Arcus Observability v2.6/New types of request tracking in Arcus Observability v2.6.md

```diff
@@ -0,0 +1,93 @@
+    }
+}
+```
+
+We also provide overloads that take in an additional Azure EventHubs consumer group and an operation name which provides a functional description of the event.
+
+## Azure Functions (isolated) HTTP trigger request tracking
+With the arrival of isolated Azure Functions, processing HTTP requests has changed. Previously, we were able to re-use the ASP.NET Core `HttpRequest`/`IActionResult` but the isolated HTTP trigger variant stepped away from the reusable approach and uses new types: `HttpRequestData`/`HttpResponseData`. This impact how we track events with our Observability library as there is no default conversion available between the two approaches.
+
+Therefore, Observability v2.6 introduces a new set of `LogRequest` overloads in the `Arcus.Observability.Telemetry.AzureFunctions` package that uses `HttpRequestData` instead of `HttpRequest`.
+
+```csharp
+using System.Net;
+using Microsoft.Extensions.Logging;
+
+public async Task<HttpResponseData> Run(
+    [HttpTrigger(AuthorizationLevel.Function, "post", Route = "v1/order")] HttpRequestData request,
+    FunctionContext executionContext)
+{
+    var statusCode = default(HttpStatusCode);
+    using (var measurement = DurationMeasurement.Start())
+    {
+        try
+        {
+            // Processing...
+            statusCode = HttpStatusCode.Ok;
+            return request.CreateResponse(statusCode)
+        }
+        finally
+        {
+            logger.LogRequest(request, statusCode, measurement);
+            // Output: {"RequestMethod": "POST", "RequestHost": "http://localhost:5000/", "RequestUri": "http://localhost:5000/v1/order", "ResponseStatusCode": 200, "RequestDuration": "00:00:00.0191554", "RequestTime": "03/23/2020 10:12:55 +00:00", "Context": {}}
+        }
+    }
+}
+```
+
+Our [Templates library](https://templates.arcus-azure.net/) will receive an update that provide you with an Azure Functions (isolated) HTTP trigger project template which will use this request tracking out-of-the-box. Note that isolated Azure Functions are able to work with custom middleware components, so the request tracking will (just like regular API applications) run in the background and not pollute your application code like with in-process Azure Functions.
+
+## Custom request tracking
+Adding these new types of request tracking got us thinking that people should be able to extend this. We provided already a way to track custom dependencies, which is in fact a way to support systems that aren't built-in supported with our Observability library. Before v2.6, we didn't provide such an extension for custom requests. With this extension, consumers can fully extend and use our library in any system possible. And because service-to-service correlation relies on the request/dependency relationship we can say with confidence that we are able to provide service correlation to any system.
```

Comment:
```suggestion
Adding these new types of request tracking got us thinking that people should be able to extend this. We already provided a way to track custom dependencies, which is in fact a way to support systems that aren't built-in supported with our Observability library. Before v2.6, we didn't provide such an extension for custom requests. With this extension, consumers can fully extend and use our library in any system possible. And because service-to-service correlation relies on the request/dependency relationship we can say with confidence that we are able to provide service correlation to any system.
```